Specifications/catalog/catalog.binding.https.md

```diff
@@ -0,0 +1,134 @@
+| reasons | Array[object] | An optional array of implementation-specific error objects. |
+
+### 2.3 The `catalog/request` endpoint
+
+#### 2.3.1 POST
+
+The [CatalogRequestMessage](catalog.protocol.md#1-catalogrequestmessage) corresponds to `POST https://<base>/catalog/request`:
+
+```
+POST https://provider.com/catalog/request
+
+Authorization: ...
+
+{
+  "@context": {
+    "ids": "https://idsa.org/"
+  },
+  "@type": "ids:CatalogRequest"
+  "ids:filter": {}
+}
+```
+
+The `Authorization` header is optional if the catalog service does not require authorization. If present, the contents of the `Authorization` header are detailed in the
+[Authorization section](#authorization).
+
+The `filter` property is optional. If present, the `filter` property can contain an implementation-specific filter expression or query to be executed as part of the catalog
+request.
+
+#### 2.3.2 OK (200) Response
+
+If the request is successful, the catalog service must return a response body containing a [CatalogMessage](./message/catalog.message.json) which is a profiled DCAT Catalog type
+described by the [Catalog Protocol Specification](catalog.protocol.md).
+
+## 3 Technical Considerations
+
+### 3.1 Authorization
+
+A catalog service may require authorization. If the catalog service requires authorization, requests must include an HTTP `Authorization` header with a token. The contents of  
+the token are undefined by may be an OAUTH2, Web DID, or other access token type.
+
+### 3.2 Versioning
+
+- Versioning will be done via URLs. TBD.
+
+### 3.3 Pagination
+
+A catalog service may paginate the results of a `CatalogRequestMessage`. Pagination data is specified using [Web Linking](https://datatracker.ietf.org/doc/html/rfc5988)
+and the HTTP `Link` header. The `Link` header will contain URLs for navigating to previous and subsequent results. The following request sequence demonstrates pagination:
+
+```
+Link: <https://provider.com/catalog?page=2&per_page=100>; rel="next"
+{
+  "@context": {
+    "dcat": "http://www.w3.org/ns/dcat/"
+  },
+  "@type": "dcat:Catalog"
+  ...
+}
+
+```
+
+Second page response:
+
+```
+Link: <https://provider.com/catalog?page=1&per_page=100>; rel="previous"
+Link: <https://provider.com/catalog?page=3&per_page=100>; rel="next"
+
+{
+   "@type": "dcat:Catalog"
+   ...
+}
+```
+
+Last page response:
+
+```
+Link: <https://provider.com/catalog?page=2&per_page=100>; rel="previous"
+
+{
+   "@type": "dcat:Catalog"
+   ...
+}
+```
+
+### 3.4 Compression
+
+Catalog services MAY compress responses to a catalog request by setting the `Content-Encoding` header to `gzip` as described in
+the [HTTP 1.1 Specification](https://www.rfc-editor.org/rfc/rfc9110.html#name-gzip-coding).
+
+## 4 Notes
+
+### 4.1 Asynchronous Interactions
+
+We may want to specify optional support for asynchronous callbacks for the catalog response. This would require addling a `callbackAddress` property and an `@id` to the request:
```

Comment:
The `@id` attribute is always there even if it doesn't look like it. If omitted (making it a BlankNode), a random one is used implicitly. Better let's make `@id` mandatory.